data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/EqualOperator.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class EqualOperator implements Operator<Boolean> {
```

Comment:
How do we want to handle when the user tries to use this operator to compare floats? Is there a worry that the precision will consider two essentially equal floats not equal? Should we consider floats equal if they are within a certain threshold like `.0001`?

Reply:
This equal operator is to cover generic object equal, which means it completely delegates to Object.equals method implementation. The case you are mentioning is valid but is conceptually different as it is an approximate equal only applies to float number type. IMO, user is not recommended to base equality off two float numbers. But if we do want to support that use case, we should define a different operator in the antlr grammar that also takes in the delta as input. Then we will have sth like ApproxOperator.eval(Object...args) that takes in 3 args.

Reply:
I think it is ok to not cover that use case at this time.